static-site/src/sections/community-page.jsx

```diff
@@ -63,11 +83,48 @@ class Index extends React.Component {
           </splashStyles.CenteredFocusParagraph>
         </FlexCenter>
         <HugeSpacer />
-        <Cards cards={communityCards}/>
         <HugeSpacer />
+        <DatasetSelect
+          title="Search Community Datasets and Narratives "
+          datasets={tableData}
+          columns={tableColumns}
+          rowSort={[]}
+        />
       </GenericPage>
     );
   }
 }
 
 export default Index;
+
+
+function parseTableData(yamlData) {
+  return yamlData.data
+    .map((entry) => {
+      /* does the URL look like a valid community URL structure? We modify this in-place
+      to remove a leading nextstrain.org domain name, as needed */
+      let url = entry.url.replace(/.*nextstrain.org/, '');
+      if (!url.startsWith("/")) url = `/${url}`;
+      const urlFields = url.split("/")
+        .slice(1); // remove first empty entry, as we ensure a leading slash
+      if (
+        urlFields.length < 3 || // minimum community URL is /community/org/repo
+        urlFields[0] !== "community" ||
+        (urlFields[1] === "narratives" && urlFields.length < 4) // narratives URLs need at least 4 parts
+      ) {
+        console.warn(`Removing data entry "${entry.name}" as URL is not valid`);
+        return undefined;
+      }
+      const d = {...entry};
+      d.url = url;
+      d.githubOrg = d.isNarrative ? urlFields[2] : urlFields[1];
+      d.githubRepo = d.isNarrative ? urlFields[3] : urlFields[2];
+      /* is the entry a dataset or a narrative?
+      NOTE that we cannot currently distinguish if a URL /community/org/repo
+      is a "default" dataset or loads the splash page. In hindsight I wish I hadn't
+      implemented "defaults" but that's what I did.             james 2022 */
+      d.isNarrative = urlFields[1]==="narratives";
+      return d;
```

Comment:
The URL and path handling here could be considerably more robust¹ and a bit simpler by constructing a [`URL` object](https://developer.mozilla.org/en-US/docs/Web/API/URL) and using pattern matching instead of doing replace/split-based string parsing. For example, something like this:

```js
const communityUrlPattern = new RegExp("^/community(?<narrative>/narratives)?/(?<org>[^/]+)/(?<repo>[^/]+)");

const url = new URL(entry.url);
const urlMatches = url.pathname.matches(communityUrlPattern);

if (!urlMatches) {
  console.warn(`Removing data entry "${entry.name}" as URL <${entry.url}> is not valid`);
  return;
}

return {
  ...entry,
  url: url.pathname,
  isNarrative: !!urlMatches.groups.narrative,
  githubOrg: urlMatches.groups.org,
  githubOrg: urlMatches.groups.repo,
};
```

If that looks better to you too, then I'll push a commit to the branch and test it.

¹ For example, a URL like https://nextstrain.org/community/tsibley/nextstrain.org/… would be handled incorrectly.

Reply:
> If that looks better to you too, then I'll push a commit to the branch and test it.

👍

Reply:
Pushed [9095e41](https://github.com/nextstrain/nextstrain.org/pull/512/commits/9095e41412d480be14e671c4893c7a61ec933f95).